dali/operators/reader/loader/video/video_test_base.cc

```diff
@@ -144,4 +144,15 @@ void VideoTestBase::SaveFrame(
   testing::SaveImage(full_path.c_str(), tv);
 }
 
+void VideoTestBase::RunFailureTest(std::function<void()> body, std::string expected_error) {
+  try {
+    body();
+
+    FAIL();   // If we reached this point body did not throw an exception.
+  } catch (const std::exception &e) {
+    EXPECT_TRUE(
+      strstr(e.what(), expected_error.c_str()));
+  }
+}
+
```

Comment:
This was moved from FramesDecoders tests, so it can be reused in more video tests.